database/sql/src/main/scala/cromwell/database/slick/tables/CallCachingAggregationEntryComponent.scala

```diff
@@ -50,6 +50,25 @@ trait CallCachingAggregationEntryComponent {
     } yield ()).exists
   )
 
+  val existsCallCachingEntriesForBaseAggregationHashWithCallCachePrefix = Compiled(
+    (baseAggregation: Rep[String],
+     prefix1: Rep[String], prefix1Length: Rep[Int],
+     prefix2: Rep[String], prefix2Length: Rep[Int],
+     prefix3: Rep[String], prefix3Length: Rep[Int]
+    ) => (for {callCachingEntry <- callCachingEntries
+      callCachingEntry <- callCachingEntries
+      if callCachingEntry.allowResultReuse
+      callCachingAggregationEntry <- callCachingAggregationEntries
+      if callCachingEntry.callCachingEntryId === callCachingAggregationEntry.callCachingEntryId
+      if callCachingAggregationEntry.baseAggregation === baseAggregation
+      detritus <- callCachingDetritusEntries
+      if detritus.callCachingEntryId === callCachingEntry.callCachingEntryId
+      detritusPath = detritus.detritusValue.map { x => x.asColumnOf[String] }
+      if (detritusPath.substring(0, prefix1Length) === prefix1) ||
+        (detritusPath.substring(0, prefix2Length) === prefix2) ||
+        (detritusPath.substring(0, prefix3Length) === prefix3)} yield ()).exists
```

Comment:
ToL: I wonder if we shouldn't start hashing the output bucket and store it along the other cache hashes so we can use it in the future directly in the hash lookup instead of having to look here, but maybe for later

Reply:
So we would add an element to our set of hashes? Wouldn't this cause misses on the existing cache hits?

Reply:
Yeah that's why I was saying "in the future" but you're right it would require a migration for existing hits.
Currently we have the `CALL_CACHING_AGGREGATION_ENTRY` table which has a "base hash" (which is everything but input files) and "input files hash".
One way would _maybe_  to add a third nullable "hint hash", that we would start populating now with an md5 of the root bucket.
We could use it by hashing the "hint buckets" we get from workflow options and filter rows for which the "hint hash" is either null or matches one of the hint buckets we have.
Again I'm really just ToL and this is likely an over optimization for now, and not a request to change this PR :)
I just think we should generally try keeping the answer to the "is this a cache hit ?" question as efficient as possible, since it has proven to be pathologically bad in the past (which this is of course very far from being).